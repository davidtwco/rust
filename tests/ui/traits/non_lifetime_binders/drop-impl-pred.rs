//@ revisions: no yes

// Issue 110557

#![feature(non_lifetime_binders)]
//~^ WARN the feature `non_lifetime_binders` is incomplete

pub trait Foo {}

#[cfg(no)]
struct Bar<T>(T) where T: Foo;

#[cfg(yes)]
struct Bar<T>(T) where for<H> H: Foo;

impl<T> Drop for Bar<T>
//[yes]~^ ERROR type annotations needed
//[no]~^^ ERROR type annotations needed
where
    for<H> H: Foo,
//[no]~^ ERROR `Drop` impl requires `H: Foo` but the struct it is implemented for does not
{
    fn drop(&mut self) {}
}

fn main() {}
