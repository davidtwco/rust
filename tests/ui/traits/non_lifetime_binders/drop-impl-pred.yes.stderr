warning: the feature `non_lifetime_binders` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/drop-impl-pred.rs:5:12
   |
LL | #![feature(non_lifetime_binders)]
   |            ^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #108185 <https://github.com/rust-lang/rust/issues/108185> for more information
   = note: `#[warn(incomplete_features)]` on by default

error[E0282]: type annotations needed
  --> $DIR/drop-impl-pred.rs:16:18
   |
LL | impl<T> Drop for Bar<T>
   |                  ^^^^^^ cannot infer type for struct `Bar<T>`

error: aborting due to 1 previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0282`.
